pkg/network/deviceinfo/deviceinfo_test.go

```diff
@@ -0,0 +1,165 @@
+		deviceInfoPlugin    = "deviceinfo"
+		nonDeviceInfoPlugin = "non_deviceinfo"
+	)
+
+	networkStatusWithMixedNetworksFmt := `[
+		{
+		    "name": "kindnet",
+		    "interface": "eth0",
+		    "ips": [
+		      "10.244.1.9"
+		    ],
+		    "mac": "3a:7e:42:fa:37:c6",
+            "default": true,
+		    "dns": {}
+	    },
+	    {
+		    "name": "default/nad1",
+		    "interface": "%s",
+		    "mac": "8a:37:d9:e7:0f:18",
+		    "dns": {}
+	    },
+	    {
+		    "name": "default/nad2",
+		    "interface": "%s",
+		    "dns": {},
+		    "device-info": {
+		      "type": "pci",
+		      "version": "1.0.0",
+		      "pci": {
+			    "pci-address": "0000:65:00.2"
+		      }
+		    }
+	    }
+    ]`
+
+	networkStatusWithPrimaryInterfaceOnly := `[
+	    {
+		    "name": "kindnet",
+		    "interface": "eth0",
+		    "ips": [
+		      "10.244.2.131"
+		    ],
+		    "mac": "82:cf:7c:98:43:7e",
+		    "default": true,
+		    "dns": {}
+	    }
+    ]`
+
+	bindingPlugins := map[string]virtv1.InterfaceBindingPlugin{
+		deviceInfoPlugin:    {DownwardAPI: virtv1.DeviceInfo},
+		nonDeviceInfoPlugin: {},
+	}
+
+	DescribeTable("should return an error",
+		func(networkStatusAnnotationValue string) {
+			networks := []virtv1.Network{*libvmi.MultusNetwork("foo", "default/nad1")}
+			interfaces := []virtv1.Interface{newBindingPluginInterface("foo", deviceInfoPlugin)}
+			_, err := deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, interfaces, networkStatusAnnotationValue, bindingPlugins)
+			Expect(err).To(HaveOccurred())
+		},
+		Entry("when networkStatus annotation is empty", ""),
+		Entry("when networkStatus annotation has invalid format", "invalid"),
+	)
+
+	DescribeTable("should prepare empty network device info annotation",
+		func(networkList []virtv1.Network, interfaceList []virtv1.Interface, networkStatusAnnotationValue string) {
+			networkDeviceInfoMap, err :=
+				deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networkList, interfaceList, networkStatusAnnotationValue, bindingPlugins)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(networkDeviceInfoMap).To(BeEmpty())
+		},
+		Entry("when there is no interface with device info binding plugin",
+			[]virtv1.Network{
+				*virtv1.DefaultPodNetwork(),
+				*libvmi.MultusNetwork("foo", "default/nad1"),
+			},
+			[]virtv1.Interface{
+				libvmi.InterfaceDeviceWithMasqueradeBinding(),
+				newBindingPluginInterface("foo", nonDeviceInfoPlugin),
+			},
+			fmt.Sprintf(networkStatusWithMixedNetworksFmt, booHashedIfaceName, fooHashedIfaceName),
+		),
+		Entry("when the interface is not in the multus status",
+			[]virtv1.Network{*libvmi.MultusNetwork("notfoo", "default/nad1")},
+			[]virtv1.Interface{newBindingPluginInterface("notfoo", deviceInfoPlugin)},
+			networkStatusWithPrimaryInterfaceOnly,
+		),
+	)
+
+	DescribeTable("should prepare network device info annotation"+
+		" when given Interfaces{1X primary- no device info, 1X no device info, 1X with device info, 1X not in multus status}",
```

Comment:
I do not think it is useful to describe the matrix here.
I suggest to try and give it a general name.

Reply:
done.